lib/request/update_helpers.js

```diff
@@ -19,7 +19,7 @@ exports.getUpdate = function (type, id, updates, cache) {
   update = { id: id }
   if (!updates[type]) updates[type] = []
   updates[type].push(update)
-  cache[type] = {}
+  cache[type] = cache[type] || {}
```

Comment:
This is actually a good catch!

I only have a nitpick, I would prefer if it was written like:

```js
if (!cache[type]) cache[type] = {}
```

This style is more consistent with adjacent code.